observation_portal/observations/admin.py

```diff
@@ -21,13 +21,31 @@ class ObservationAdmin(admin.ModelAdmin):
     raw_id_fields = ('request', )
     form = ObservationForm
     inlines = [ConfigurationStatusInline]
+    change_form_template = "observations/observation_changeform.html"
+
+    def response_change(self, request, obj):
+        if "refund-time" in request.POST:
+            refund_percent = int(request.POST['refund-percent'])
+            time_refunded = refund_observation_time(obj, refund_percent / 100.0)
+            messages.info(request, f"Refunded {time_refunded} hours back to the TimeAllocation")
+            return HttpResponseRedirect(".")
+        return super().response_change(request, obj)
 
 
 class ConfigurationStatusAdmin(admin.ModelAdmin):
     list_display = ('id', 'state', 'instrument_name', 'guide_camera_name')
     list_filter = ('state', )
     raw_id_fields = ('observation', 'configuration')
     inlines = [SummaryInline]
+    change_form_template = "observations/observation_changeform.html"
+
+    def response_change(self, request, obj):
+        if "refund-time" in request.POST:
+            refund_percent = int(request.POST['refund-percent'])
```

Comment:
Might be overkill but should we do some validation on the percentage ([0, 100])? Though I see you did it in the html form

Reply:
since its in the form I don't think its possible to submit the form with a number out of that range...